File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/lobby/NewPlayer.java
package tech.zolhungaj.amqapi.servercommands.gameroom.lobby;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAvatar;

import java.util.Optional;

public record NewPlayer(

	@Json(name = "level")
	int level,

	@Json(name = "ready")
	Boolean ready,

	@Json(name = "name")
	String playerName,

	@Json(name = "teamNumber")
	Optional<Integer> teamNumber,

	@Json(name = "gamePlayerId")
	int gamePlayerId,

	@Json(name = "avatar")
	PlayerAvatar avatar,

	@Json(name = "inGame")
	Boolean inGame
) implements Command {
	@Override
	public String commandName() {
		return CommandTypeOld.NEW_PLAYER.commandName;
	}
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/lobby/NewPlayer.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.lobby

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAvatar

@JvmRecord
@CommandType("New Player")
data class NewPlayer(
    @Json(name = "level")
    val level: Int,
    @Json(name = "ready")
    val ready: Boolean,
    @Json(name = "name")
    val playerName: String,
    @Json(name = "teamNumber")
    val teamNumber: Int?,
    @Json(name = "gamePlayerId")
    val gamePlayerId: Int,
    @Json(name = "avatar")
    val avatar: PlayerAvatar,
    @Json(name = "inGame")
    val inGame: Boolean
) 